src/patternfly/components/ProgressStepper/progress-stepper.scss

```diff
@@ -0,0 +1,248 @@
+    --pf-c-progress-stepper--GridTemplateColumns: var(--pf-c-progress-stepper--m-compact--GridTemplateColumns);
+    --pf-c-progress-stepper__step-icon--Width: var(--pf-c-progress-stepper--m-compact__step-icon--Width);
+    --pf-c-progress-stepper__step-icon--FontSize: var(--pf-c-progress-stepper--m-compact__step-icon--FontSize);
+    --pf-c-progress-stepper__pficon--MarginTop: var(--pf-c-progress-stepper--m-compact__pficon--MarginTop);
+    --pf-c-progress-stepper__fa-exclamation-triangle--MarginTop: var(--pf-c-progress-stepper--m-compact__fa-exclamation-triangle--MarginTop);
+    --pf-c-progress-stepper__step-main--MarginTop: var(--pf-c-progress-stepper--m-compact__step-main--MarginTop);
+    --pf-c-progress-stepper__step-main--MarginBottom: var(--pf-c-progress-stepper--m-compact__step-main--MarginBottom);
+    --pf-c-progress-stepper__step-title--FontSize: var(--pf-c-progress-stepper--m-compact__step-title--FontSize);
+    --pf-c-progress-stepper__step--m-current__step-title--FontWeight: var(--pf-c-progress-stepper--m-compact__step-title--FontWeight);
+
+    .pf-c-progress-stepper__step:not(.pf-m-current) .pf-c-progress-stepper__step-main {
+      @include pf-u-screen-reader;
+    }
+
+    .pf-c-progress-stepper__step-description {
+      display: none;
+    }
+
+    // For this compact variant only, move the main content for the current step to above the icons
+    // stylelint-disable
+    .pf-c-progress-stepper__step.pf-m-current .pf-c-progress-stepper__step-main {
+        grid-column: 1/-1;
+        grid-row: 1/2;
+      }
+    // stylelint-enable
+  }
+}
+
+// Step
+.pf-c-progress-stepper__step {
+  display: contents;
+
+  // Step Modifiers
+  &.pf-m-current {
+    --pf-c-progress-stepper__step-title--FontWeight: var(--pf-c-progress-stepper__step--m-current__step-title--FontWeight);
+    --pf-c-progress-stepper__step-title--Color: var(--pf-c-progress-stepper__step--m-current__step-title--Color);
+  }
+
+  &.pf-m-pending {
+    --pf-c-progress-stepper__step-title--Color: var(--pf-c-progress-stepper__step--m-pending__step-title--Color);
+  }
+
+  &.pf-m-success {
+    --pf-c-progress-stepper__step-icon--Color: var(--pf-global--success-color--100);
+  }
+
+  &.pf-m-danger {
+    --pf-c-progress-stepper__step-icon--Color: var(--pf-global--danger-color--100);
+    --pf-c-progress-stepper__step-title--Color: var(--pf-c-progress-stepper__step--m-danger__step-description--Color);
+  }
+
+  &.pf-m-warning {
+    --pf-c-progress-stepper__step-icon--Color: var(--pf-global--warning-color--100);
+  }
+
+  &.pf-m-info {
+    --pf-c-progress-stepper__step-icon--Color: var(--pf-global--info-color--100);
+  }
+}
+
+// The step icon wrapper provides a container on which to use the before for the line connecting the steps
+.pf-c-progress-stepper__step-connector {
+  position: relative;
+  display: flex;
+  justify-content: var(--pf-c-progress-stepper__step-connector--JustifyContent);
+  width: 100%;
+  transform: translateX(0);
```

Comment:
It looks like this is to create a new local coordinate system so the negative z-index on the `::before` works? If so, you could also just give the icon a position z-index using the sm variable. That's how we typically do that sort of thing.